vault/token_store.go

```diff
@@ -1193,6 +1193,13 @@ func (ts *TokenStore) handleCreateCommon(
 		return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
 	}
 
+	// Prevent internasl policies from being assigned to tokens
+	for _, policy := range te.Policies {
+		if strutil.StrListContains(nonAssignablePolicies, policy) {
+			return logical.ErrorResponse(fmt.Sprintf("cannot assign %s policy", policy)), nil
```

Comment:
I'm on the fence whether the right thing to do is simply filter them out, or return an error. Returning an error does force fixing bad code, but in other places we manipulate the asked-for list (e.g. to add `default`).